.circleci/config.yml

```diff
@@ -44,6 +44,9 @@ aliases:
       install_swiftlint:
         type: boolean
         default: true
+      dry_run:
+        type: boolean
+        default: false
     environment:
```

Comment:
this bit I don't _love_: I had to make it a parameter of _base job_ rather than the concrete deploy purchase tester job. 

It doesn't actually affect anything other than the fact that all jobs get this param now. Maybe that's deceiving in that if you run the other deploy stuff with `dry_run: true`, it's not an actual dry run, and that's a bit concerning. 

But I couldn't find a better way of keeping it specific to the job: the thing is that we inherit all the stuff from the base job, and that includes `parameters`. If I declare it again, it breaks the config. 

I could also not inherit from base job and just copy/paste all of its things, but I feel like that's worse than what we had before. 

I'm open to other ideas, especially since overall this doesn't entirely feel better to me than having two almost identical jobs. I'm worried that a future dev could think that `dry_run` on all jobs actually means dry run.

Reply:
Right, I certainly don't love it either and it can be confusing... Not aware of any other options off the top of my head, so I agree with you we might as well leave it as 2 separate jobs.

Reply:
I think the base job should actually be an executor not an alias:

```
executors:
  macos-executor:
    parameters:
      xcode_version:
        type: string
        default: '15.3'
      install_swiftlint:
        type: boolean
        default: true
    macos:
      xcode: << parameters.xcode_version >>
    resource_class: macos.m1.medium.gen1
    environment:
      CIRCLECI_TESTS_GENERATE_SNAPSHOTS: << pipeline.parameters.generate_snapshots >>
      CIRCLECI_TESTS_GENERATE_REVENUECAT_UI_SNAPSHOTS: << pipeline.parameters.generate_revenuecatui_snapshots >>
    working_directory: ~/purchases-ios
    shell: /bin/bash --login -o pipefail
```

We can keep the alias so we don't have to update all jobs now:

```
aliases:
  base-job: &base-job
    executor:
      name: macos-executor
```

Then you can have custom parameters for the job

```
 deploy-purchase-tester:
    executor:
      name: macos-executor
    parameters:
      dry_run:
        type: boolean
        default: false
```

Reply:
ohh that's great, I'll give it a shot. Thanks!

Reply:
Updated with this approach, looks soooo much better! I hope that it does work. If it does I'll do another follow-up to remove the old base job alias

Reply:
I ran into some issues with failing tests when mixing and matching this approach and the alias... And this approach is just better than the alias, so I went ahead and replaced it altogether